File: Homework/hw2-OOP/beast.cpp
/**
 * beast.cpp - contains functions for working with beast
 */

#include <cstring>
#include "beast.h"
#include "rnd.h"

// Enter beast parameters form file
void Beast::in(FILE *file){
    int type;
    fscanf(file,"%s", name_);
    fscanf(file,"%i", &weight_);
    fscanf(file,"%i", &type);

    if(type == 0){
        type_ = CARNIVORE;
    }else if(type == 1){
        type_ = HERBIVORE;
    }else if(type == 2){
        type_ = OMNIVORE;
    }
}

// Enter random parameters for beast
void Beast::inRnd(){
    type_ = static_cast<BeastEnum>(randomInt(3) % 3);
    weight_ = randomInt(2000);
    strcpy(name_, "BeastName");
}

// Output fish parameters In a file stream
void Beast::out(FILE *file){
    char* type;

    switch (type_) {
        case 0:
            type = "carnivore";
            break;
        case 1:
            type = "herbivore";
            break;
        case 2:
            type = "omnivore";
            break;
    }

    fprintf(file, "It is a Beast: name = %s, weight: %i, is: %s, and has a special number: %f\n",
            name_, weight_, type, specialNumber());
}

double Beast::specialNumber(){
    double charSum = 0;
    for (int i = 0 ; i < sizeof(name_) / sizeof(name_[0]) ; ++i) {
        charSum += name_[i];
    }
    return charSum / weight_;
}

File: Homework/hw3-python/beast.py
import animal as animal

from animal import *


# ----------------------------------------------
class Beast(Animal):
    def __init__(self):
        super().__init__()
        self.type = 1

    def read_str_array(self, strArray, i):
        # должно быт как минимум три непрочитанных значения в массиве
        if i >= len(strArray) - 2:
            return 0
        self.name = strArray[i]
        self.weight = int(strArray[i + 1])
        self.type = int(strArray[i + 2])
        i += 3
        return i

    def print(self):
        animal_type = ""

        if self.type == 1:
            animal_type = "Carnivore"
        elif self.type == 2:
            animal_type = "Herbivore"
        elif self.type == 3:
            animal_type = "Omnivore"

        print("Beast: name = ", self.name, " weight = ", self.weight, "type = ", animal_type,
              ", special_number = ", self.special_number())

    def write(self, ostream):
        animal_type = ""

        if self.type == 1:
            animal_type = "Carnivore"
        elif self.type == 2:
            animal_type = "Herbivore"
        elif self.type == 3:
            animal_type = "Omnivore"

        ostream.write("Beast: name = {}  weigh = {}  type = {}, special_number = {}"
                      .format(self.name, self.weight, animal_type, self.special_number()))
